consensus/tendermint/backend/engine_test.go

```diff
@@ -453,22 +458,253 @@ func TestClose(t *testing.T) {
+
+		for i := 0; i < stop; i++ {
+			wg.Add(1)
+
+			go func(wg *sync.WaitGroup) {
+				defer wg.Done()
+
+				err := b.Close()
+				errC <- err
+			}(&wg)
+
+		}
+
+		wg.Wait()
+		close(errC)
+
+		assertNotCoreStarted(t, b)
+
+		var sawNil bool
+
+		// Want nil once and ErrStoppedEngine 9 times
+		for e := range errC {
+			if e == nil {
+				if sawNil {
+					t.Fatalf("<nil> returned more than once, b.Close() should have only returned nil the first time it was closed")
+				}
+				sawNil = true
+			} else if e != ErrStoppedEngine {
+				assertError(t, ErrStoppedEngine, e)
+			}
+		}
+	})
+}
+
+func TestStart(t *testing.T) {
+	t.Run("engine is not running, no errors", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		ctx := context.Background()
+		tendermintC := tendermintCore.NewMockTendermint(ctrl)
+		tendermintC.EXPECT().Start(ctx).Return(nil).MaxTimes(1)
+
+		b := &Backend{
+			core:        tendermintC,
+			coreStarted: false,
+		}
+
+		err := b.Start(ctx, &core.BlockChain{}, func() *types.Block { return &types.Block{} }, func(hash common.Hash) bool { return false })
+		assertCoreStarted(t, b)
+		assertNilError(t, err)
+	})
+
+	t.Run("engine is running, error returned", func(t *testing.T) {
+		b := &Backend{
+			coreStarted: true,
+		}
+
+		err := b.Start(context.Background(), &core.BlockChain{}, func() *types.Block { return &types.Block{} }, func(hash common.Hash) bool { return false })
+		assertCoreStarted(t, b)
+		assertError(t, ErrStartedEngine, err)
 	})
+
+	t.Run("engine is not running, started twice", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		ctx := context.Background()
+		tendermintC := tendermintCore.NewMockTendermint(ctrl)
+		tendermintC.EXPECT().Start(ctx).Return(nil).MaxTimes(1)
+
+		b := &Backend{
+			core:        tendermintC,
+			coreStarted: false,
+		}
+
+		err := b.Start(ctx, &core.BlockChain{}, func() *types.Block { return &types.Block{} }, func(hash common.Hash) bool { return false })
+		assertCoreStarted(t, b)
+		assertNilError(t, err)
+
+		err = b.Start(ctx, &core.BlockChain{}, func() *types.Block { return &types.Block{} }, func(hash common.Hash) bool { return false })
+		assertCoreStarted(t, b)
+		assertError(t, ErrStartedEngine, err)
+	})
+
+	t.Run("engine is not running, started multiple times", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		ctx := context.Background()
+		tendermintC := tendermintCore.NewMockTendermint(ctrl)
+		tendermintC.EXPECT().Start(ctx).Return(nil).MaxTimes(1)
+
+		b := &Backend{
+			core:        tendermintC,
+			coreStarted: false,
+		}
+
+		var wg sync.WaitGroup
+		stop := 10
+		errC := make(chan error, stop)
+
+		for i := 0; i < stop; i++ {
+			wg.Add(1)
+
+			go func(wg *sync.WaitGroup) {
+				defer wg.Done()
+
+				err := b.Start(ctx, &core.BlockChain{}, func() *types.Block { return &types.Block{} }, func(hash common.Hash) bool { return false })
+				errC <- err
```

Comment:
```suggestion
				errC <- b.Start(ctx, &core.BlockChain{}, func() *types.Block { return &types.Block{} }, func(hash common.Hash) bool { return false })
```

There's no need to create a variable here.

Reply:
Ok I will update